### PR TITLE
Bump GDAL to 3.5.3 and rasterio to 1.3.3

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [ ] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

- Newer versions of GDAL, rasterio

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: data-api-gdal.dockerfile [hunk 1]
```diff
@@ -10,9 +10,9 @@ RUN pip3 install \
...
         awscli~=1.25.44 \
         fiona~=1.8.21 \
-        rasterio~=1.3.0 \
+        rasterio~=1.3.3 \
         boto3~=1.24.44 \
         awscli-plugin-endpoint~=0.4 \
         tileputty~=0.2.8
 
-RUN ln -s /usr/include /usr/include/gdal
+RUN ln -s /usr/include /usr/include/gdal
```
Comment (gtempus) [inline data-api-gdal.dockerfile:18]: Did you mean to remove the newline?

File: data-api-gdal.dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
-#1.1.7
-FROM osgeo/gdal:ubuntu-small-3.5.1
+#1.1.8
```
Comment (gtempus) [inline data-api-gdal.dockerfile:1]: I'm guessing that this is our own internal version?
Would a [docker `LABEL`](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#label) be helpful in this instance? Just wondering.

**Review by gtempus — COMMENTED** [review]
Thanks, @dmannarino!

File: data-api-gdal.dockerfile (hunk 1 above)
Comment (dmannarino) [reply data-api-gdal.dockerfile:19]: Yeah, doesn't really matter though.

File: data-api-gdal.dockerfile (hunk 2 above)
Comment (dmannarino) [reply data-api-gdal.dockerfile]: Yep! How's this? https://github.com/wri/gfw-dockerfiles/pull/5/commits/a338b7a87e015025a4a0aa03eddd5241bcf55722

**Pull request merged**